### PR TITLE
CS-2090 / CS-2094 / CS-2096 / CS-2146 / CS-2095

### PR DESCRIPTION
**Pull request by jonathanweyermann**
image
image

image

### CODE REVIEW

**Review by rstyan — CHANGES_REQUESTED** [review]

File: apps/status-service/src/app/jobs/checkEndpoint.ts [hunk 1]
```diff
@@ -90,12 +112,85 @@ export const getNewEndpointStatus = (
...
+
+      const waitTimeHistoryDouble = await endpointStatusEntryRepository.findRecentByUrlAndApplicationId(
+        app.url,
+        app.appKey,
+        waitTime
+      );
+
+      let switchOffCount = 0;
+      let switchOnCount = 0;
+      waitTimeHistoryDouble.forEach((w, index) => {
```
[the remark this answers is not in the document]
Comment (jonathanweyermann) [reply apps/status-service/src/app/jobs/checkEndpoint.ts:159]: ok

File: apps/status-service/src/app/jobs/checkEndpoint.ts [hunk 2]
```diff
@@ -90,12 +112,85 @@ export const getNewEndpointStatus = (
...
+
+  const webhooks = response.data.latest.configuration;
+
+  Object.keys(webhooks).map(async (key) => {
+    const webhook = webhooks[key] as Webhooks;
+    if (webhook.targetId === app.appKey) {
+      const eventTypes = webhook.eventTypes;
+      const waitTime = 2 * (webhook.intervalSeconds / 60);
+
+      const waitTimeHistoryDouble = await endpointStatusEntryRepository.findRecentByUrlAndApplicationId(
```
[the remark this answers is not in the document]
Comment (jonathanweyermann) [reply apps/status-service/src/app/jobs/checkEndpoint.ts:151]: ok, I'll make this more clear

[the remark this answers is not in the document]
Comment (jonathanweyermann) [reply apps/status-service/src/app/jobs/checkEndpoint.ts:151]: Its basically double the number of pings that waittime reflects

File: apps/status-service/src/app/jobs/checkEndpoint.ts [hunk 3]
```diff
@@ -90,12 +112,85 @@ export const getNewEndpointStatus = (
...
+        app.url,
+        app.appKey,
+        waitTime
+      );
+
+      let switchOffCount = 0;
+      let switchOnCount = 0;
+      waitTimeHistoryDouble.forEach((w, index) => {
+        if (w.ok && index === waitTimeHistoryDouble.length / 2) {
```
[the remark this answers is not in the document]
Comment (jonathanweyermann) [reply apps/status-service/src/app/jobs/checkEndpoint.ts:160]: I'm intentionally ignoring them. I don't require that all pings be up or down before you have waitTime number of down pings, just that there is one last up and then waitTime number of down pings. Essentially i'm just looking for waitTime number of down pings in a row, regardless of what happens before. This switchoncount is a little different and takes into account the whole row, so the logic is a little mixed in there with that, which I guess could be confusing

File: apps/push-service/src/push/router/stream.ts [hunk 4]
```diff
@@ -14,13 +23,48 @@ import { Logger } from 'winston';
...
 import { StreamEntity, StreamItem } from '../model';
 import { ExtendedError } from 'socket.io/dist/namespace';
+import axios from 'axios';
+
+import { webhookTriggered } from '../events';
 
 interface StreamRouterProps {
   logger: Logger;
-  eventService: DomainEventSubscriberService;
+  eventServiceAmp: DomainEventSubscriberService;
```
[the remark this answers is not in the document]
Comment (jonathanweyermann) [reply apps/push-service/src/push/router/stream.ts:32]: Probably have to do this regardless

**Review by rstyan — APPROVED** [review]

**Pull request merged**